mlflow/utils/gorilla.py

```diff
@@ -292,10 +293,20 @@ def apply(patch):
     """
     settings = Settings() if patch.settings is None else patch.settings
 
+    curr_active_patch = _ACTIVE_PATCH % (patch.name,)
+    if curr_active_patch in patch.destination.__dict__:
+        # safe guarding checking.
+        raise RuntimeError(
+            f"Patch {patch.name} on {destination.__name__} already existed."
+            f"Revert old patch first if you want to patch it again."
+        )
+
     # When a hit occurs due to an attribute at the destination already existing
     # with the patch's name, the existing attribute is referred to as 'target'.
     try:
-        target = get_attribute(patch.destination, patch.name)
+        target = get_original_attribute(
+            patch.destination, patch.name, bypass_descriptor_protocal=True
+        )
```

Comment:
This line changing address one bug: make patch on top of patch. Suppose the case:
```
class A:
    def predict(self):
        pass

class B(A):
    pass
```

Suppose we patch `A.predict` first, and then patch `B.predict`, when we do patching `B.predict`, the previous code `get_attribute(patch.destination, patch.name)` will get the patched `A.predict` method, so it will generate a new patch method on top of patched `A.predict`

I modify the `gorilla.get_original_attribute` impl to make it support class hierarchy correctly, and also make it support bypassing/not bypassing descriptor protocal.